xarray/core/dataarray.py

```diff
@@ -882,10 +882,46 @@ def reindex(self, method=None, tolerance=None, copy=True, **indexers):
             method=method, tolerance=tolerance, copy=copy, **indexers)
         return self._from_temp_dataset(ds)
 
+    def interp(self, method='linear', kwargs={}, assume_sorted=False,
+               **coords):
+        """ Multidimensional interpolation of variables.
+
+        **coords : {dim: new_coordinate, ...}
+            Keyword arguments with names matching dimensions and values.
+            coords can be an integer, array-like or DataArray.
+            If DataArrays are passed as coords, their dimensions are used
+            for the broadcasting.
+        method: {'linear', 'nearest'} for multidimensional array,
+            {'linear', 'nearest', 'zero', 'slinear', 'quadratic', 'cubic'}
+            for 1-dimensional array.
+        assume_sorted: boolean, optional
+            If False, values of x can be in any order and they are sorted
+            first. If True, x has to be an array of monotonically increasing
+            values.
+        kwargs: dictionary
+            Additional keyword passed to scipy's interpolator.
+
+        Returns
+        -------
+        interpolated: xr.DataArray
+            New dataarray on the new coordinates.
+
+        Note
+        ----
+        scipy is required.
+
+        See Also
+        --------
+        scipy.interpolate.interp1d
+        scipy.interpolate.interpn
+        """
+        ds = self._to_temp_dataset().interp(
+            method=method, kwargs=kwargs, **coords)
```

Comment:
should you be passing `assume_sorted` through here?

Reply:
Good catch! Thanks.
fixed.